pkg/command/service/generate.go

```diff
@@ -0,0 +1,200 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+
+	knativeapis "knative.dev/pkg/apis"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+
+	"github.com/spf13/cobra"
+
+	"knative.dev/kperf/pkg"
+	"knative.dev/kperf/pkg/generator"
+)
+
+var (
+	count, interval, batch, concurrency, minScale, maxScale int
+	nsPrefix, nsRange, ns                                   string
+	svcNamePrefixDefault                                    string = "testksvc"
+	checkReady                                              bool
+	timeout                                                 time.Duration
+	ksvcClient                                              *servingv1client.ServingV1Client
+	err                                                     error
+)
+
+func NewServiceGenerateCommand(p *pkg.PerfParams) *cobra.Command {
+	ksvcGenCommand := &cobra.Command{
+		Use:   "generate",
+		Short: "generate ksvc",
+		Long: `generate ksvc workload
+
+For example:
+# To generate ksvc workload
+kperf service generate —count 500 —interval 20 —batch 20 --min-scale 0 --max-scale 5 (--nsprefix testns/ --ns nsname)
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var nsRangeMap map[string]bool = map[string]bool{}
+			if nsPrefix != "" {
+				r := strings.Split(nsRange, ",")
+				if len(r) != 2 {
+					fmt.Printf("Expected Range like 1,500, given %s\n", nsRange)
+					os.Exit(1)
```

Comment:
Ditto same comments in this file. Although it looks like it's the same code, so you need to extract this to a library, I think.

Reply:
Make sense to me. Will add a common function for parsing, covered #4